src/Umbraco.Web/Routing/ContentFinderByUrlAlias.cs

```diff
@@ -64,11 +67,22 @@ private static IPublishedContent FindContentByAlias(IPublishedContentCache cache
 
             const string propertyAlias = Constants.Conventions.Content.UrlAlias;
 
-            var test1 = alias.TrimStart('/') + ",";
-            var test2 = ",/" + test1; // test2 is ",/alias,"
-            test1 = "," + test1; // test1 is ",alias,"
+            var hasDomain = !domaineName.IsNullOrWhiteSpace();
 
-            bool IsMatch(IPublishedContent c, string a1, string a2)
+            // Try to find any first relative path
+            var match = Regex.Match(domaineName, @"(?<!http:\/|http:|https:\/|https:)\/.+");
+
+            // Create the list of potential alias that can be found on umbracoAliasUrl
+            var trimAlias = hasDomain ? alias.TrimStart(match.Value) : alias;
```

Comment:
`TrimStart()` might remove the value multiple times, so this should be replaced with a check/substring instead.